### PR TITLE
switched to SecureStore/fixed onboarding chore and reward submission bug/android UI fixes

### PR DESCRIPTION
**Pull request by michaeldegori**
switched to SecureStore/fixed onboarding chore and reward submission bug/android UI fixes

### CODE REVIEW

File: src/routes/AddFamilyUnitMember/AddChildView.js [hunk 1]
```diff
@@ -126,7 +133,7 @@ const AddChildView = ({
 							<View style={styles.dobContainer}>
 								<View style={styles.dobInputMD}>
 									<KKTextInput
-										value={dobM}
+										value={null}
```
Comment (Swolebrain) [inline src/routes/AddFamilyUnitMember/AddChildView.js:136]: this seems broken.

File: src/routes/AddFamilyUnitMember/AddChildView.js [hunk 2]
```diff
@@ -135,7 +142,7 @@ const AddChildView = ({
 								</View>
 								<View style={styles.dobInputMD}>
 									<KKTextInput
-										value={dobD}
+										value={null}
```
Comment (Swolebrain) [inline src/routes/AddFamilyUnitMember/AddChildView.js:145]: here too

File: src/routes/AddFamilyUnitMember/AddChildView.js [hunk 3]
```diff
@@ -144,7 +151,7 @@ const AddChildView = ({
 								</View>
 								<View style={styles.dobInputY}>
 									<KKTextInput
-										value={dobY}
+										value={null}
```
Comment (Swolebrain) [inline src/routes/AddFamilyUnitMember/AddChildView.js:154]: also here

File: src/routes/Alerts/AlertsView.js [hunk 4]
```diff
@@ -37,8 +39,9 @@ const AlertView = ({
 		modalVisible={modalVisible}
 		modalClose={modalClose}
 		renderModalContents={renderModalContents(modalText, modalAccept, modalDeny)}
+		style={{ paddingTop: Platform.OS === 'android' ? getStatusBarPadding() : 0 }}
```
Comment (Swolebrain) [inline src/routes/Alerts/AlertsView.js:42]: why not put the platform conditional in the getStatusBarPadding function? Seems like this funtion is incomplete

File: src/routes/KreditDashboard/KidKreditDashboardView.js [hunk 5]
```diff
@@ -10,9 +10,14 @@ import InfoIcon from './InfoIcon';
 import EducationalModal from '../../common/EducationalModal';
 import { scaleRatio } from '../../configuration';
 import BigTextLabel from '../../common/BigTextLabel';
+import { getStatusBarPadding } from '../../common/NavBarPaddingView';
 
 const KidKreditDashboardView = ({ loading, kreditInfo, modalVisible, showModal, hideModal, educationalInfo = {} }) => (
-	<FullPage>
+	<FullPage
+		style={{
+			paddingTop: Platform.OS === 'android' ? getStatusBarPadding() : 0,
```
Comment (Swolebrain) [inline src/routes/KreditDashboard/KidKreditDashboardView.js:18]: getStatusBarHeight should do this

File: src/routes/KreditDashboard/KreditDashboardView.js [hunk 6]
```diff
@@ -44,61 +71,111 @@ const KreditDashboardView = ({
 	</FullPage>
 );
 
-const renderKidKreditInfo = (kidsList = [], showModal) => selectedChildId => {
+const renderKidKreditInfo = (
+	kidsList = [],
+	showModal,
+	showBBModal,
```
Comment (Swolebrain) [inline src/routes/KreditDashboard/KreditDashboardView.js:77]: this is very redundant. You never need a "Show" and a "hide" variable for the same thing. It seems like three variables are related to the same thing here. Only one piece of state is needed to determine whether the modal is visible or not.

File: src/routes/KreditDashboard/KreditDashboardView.js [hunk 7]
```diff
@@ -44,61 +71,111 @@ const KreditDashboardView = ({
...
+	hideBBModal,
+	bbModalVisible,
+	addBBs,
+	subtractBBs,
+	setBBs,
```
Comment (Swolebrain) [inline src/routes/KreditDashboard/KreditDashboardView.js:82]: subtractBBs and setBBs are redundant. Subtracting is just adding a negative number. We need to reduce code repetition here.

File: src/routes/KreditDashboard/index.js [hunk 8]
```diff
@@ -1,55 +1,97 @@
...
+	};
+
+	hideBBModal = () => {
+		this.setState(() => ({
+			bbModalVisible: false,
+		}));
+	};
+
+	subtractBBs = () => {
+		this.setState({ bambooBucks: (this.state.bambooBucks -= 1) });
```
Comment (Swolebrain) [inline src/routes/KreditDashboard/index.js:61]: when adding or subtracting 1, it is bad practice to not use functional setstate. Please read https://www.freecodecamp.org/news/functional-setstate-is-the-future-of-react-374f30401b6b/

File: src/routes/RewardsFeed/KidRewardsView.js [hunk 9]
```diff
@@ -36,7 +37,13 @@ const KidRewardsView = ({
 		modalVisible={modalVisible}
 		renderModalContents={renderModalContents(modalText, modalAccept, modalClose)}
 		modalClose={() => modalClose(false)}
-		style={{ flex: 1, alignSelf: 'stretch', justifyContent: 'flex-start', alignItems: 'center' }}
+		style={{
+			flex: 1,
+			alignSelf: 'stretch',
+			justifyContent: 'flex-start',
+			alignItems: 'center',
+			paddingTop: Platform.OS === 'android' ? getStatusBarPadding() : 0,
```
Comment (Swolebrain) [inline src/routes/RewardsFeed/KidRewardsView.js:45]: repetitive here too

File: src/routes/Settings/TermsOfUse.js [hunk 10]
```diff
@@ -1,32 +1,42 @@
 import React from 'react';
-import {ActivityIndicator, Dimensions, View} from 'react-native';
+import { ActivityIndicator, Dimensions, View } from 'react-native';
```
Comment (Swolebrain) [inline src/routes/Settings/TermsOfUse.js:2]: please turn off your automatic formatting or be more mindful of what you commit. These non-changes are cluttering the PR process.

File: src/routes/onboarding/OnboardingChoreSuggestionsContainer.js [hunk 11]
```diff
@@ -195,110 +197,127 @@ class OnboardingChoreSuggestionsContainer extends React.Component {
...
-		];
-
-		const kidsList = familyUnitRepository?.kidsList;
-		const existingChores = familyUnitRepository.existingChores;
-		const child = kidsList[kidsList.length - 1];
+		],
+	};
 
+	componentDidMount() {
+		this.setState({
```
Comment (Swolebrain) [inline src/routes/onboarding/OnboardingChoreSuggestionsContainer.js:204]: once again please don't add things from the repository to local state. Very bad practice and bug prone. The point of a repository is to not need local state for that thing.

File: src/routes/onboarding/OnboardingChoreSuggestionsContainer.js [hunk 12]
```diff
@@ -11,10 +11,12 @@ import choresRepository from '../../stores/DefaultChoresStore';
...
-
-	render() {
-		const defaultChores = [
+		choreList: [],
+		kidsList: [],
+		checked: [],
+		existingChores: [],
+		child: {},
+		defaultChores: [
 			{
```
Comment (Swolebrain) [inline src/routes/onboarding/OnboardingChoreSuggestionsContainer.js:20]: please move all these to a separate file.

File: src/routes/onboarding/OnboardingChoreSuggestionsContainer.js [hunk 13]
```diff
@@ -195,110 +197,127 @@ class OnboardingChoreSuggestionsContainer extends React.Component {
...
-								startDate: new Date().getTime(),
-								choreAppliedTo: child._id,
-							},
-							userRepository.idToken
-						);
-					} catch (err) {
-						Alert.alert('Error posting new chore', err.toString());
-					}
+		for (const chore of this.state.checked) {
+			const existingChore = this.state.existingChores.find(eachChore => eachChore.name === chore.name);
```
Comment (Swolebrain) [inline src/routes/onboarding/OnboardingChoreSuggestionsContainer.js:244]: is this line extraneous?

File: src/routes/onboarding/OnboardingRewardSuggestionsContainer.js [hunk 14]
```diff
@@ -67,38 +63,56 @@ class OnboardingRewardSuggestionsContainer extends React.Component {
...
-							kkCost: reward.cost,
-							notes: reward.notes,
-						},
-						userRepository.idToken
-					);
-				} catch (err) {
-					console.log(err);
-					Alert.alert('API error', typeof err === 'object' ? JSON.stringify(err, null, 2) : err.toString());
-				}
+	handleCheck = reward => {
```
Comment (Swolebrain) [inline src/routes/onboarding/OnboardingRewardSuggestionsContainer.js:73]: this is bug-prone. Bad practice to rely on current state to set new state. Read the article I posted on functional setState.

File: src/services/Authorization.js [hunk 15]
```diff
@@ -76,21 +77,19 @@ async function registerWithAuth0(email, password, firstName, lastName, parent_ty
...
 async function logOutFromAuth0(history) {
-	await AsyncStorage.multiRemove([
-		'@kiddiekredit:idToken',
-		'@kiddiekredit:accessToken',
-		'@kiddiekredit:refreshToken',
-		'@kiddiekredit:expiresIn',
-	]);
-
-	await AsyncStorage.removeItem('BROWSING_MODE');
+	await SecureStore.deleteItemAsync('kk-idToken');
```
Comment (Swolebrain) [inline src/services/Authorization.js:87]: Please use Promise.all here so these aren't all sequential.

File: src/services/MixPanelAlt.js [hunk 16]
```diff
@@ -0,0 +1,171 @@
+import ExpoMixpanelAnalytics from '@benawad/expo-mixpanel-analytics';
```
Comment (Swolebrain) [inline src/services/MixPanelAlt.js:1]: is this file redundant?

File: src/stores/UserDataStore.js [hunk 17]
```diff
@@ -107,33 +108,33 @@ class UserDataStore {
...
 			userInfo.email
 		);
-		return await AsyncStorage.multiSet([
-			['@kiddiekredit:idToken', idToken],
-			['@kiddiekredit:accessToken', accessToken],
-			['@kiddiekredit:refreshToken', refreshToken],
-			['@kiddiekredit:expiresIn', '' + Number(userInfo.exp) * 1000],
-			['@kiddiekredit:email', userInfo.email],
-		]);
+		await SecureStore.setItemAsync('kk-idToken', idToken);
```
Comment (Swolebrain) [inline src/stores/UserDataStore.js:119]: Please use Promise.all

**Review by Swolebrain — CHANGES_REQUESTED** [review]

File: src/routes/onboarding/OnboardingChoreSuggestionsContainer.js (hunk 13 above)
Comment (michaeldegori) [reply src/routes/onboarding/OnboardingChoreSuggestionsContainer.js:244]: Hmm, if it is, i'm having trouble seeing what else to do.

**Review by Swolebrain — APPROVED** [review]

**Pull request merged**